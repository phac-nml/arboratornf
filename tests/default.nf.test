nextflow_pipeline {

    name "Test pipeline"
    script "../main.nf"
    tag "pipeline"

    test("-profile test") {

        when {
            params {
                input = "$baseDir/tests/data/samplesheets/samplesheet.csv"
                outdir = "$outputDir"

                metadata_partition_name = "outbreak"
                metadata_1_header = "organism"
                metadata_2_header = "subtype"
                metadata_3_header = "country"
                metadata_4_header = "serovar"
                metadata_5_header = "age"
                metadata_6_header = "date"
                metadata_7_header = "source"
                metadata_8_header = "special"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}', 'arborator/run.json', 'iridanext.output.json'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we test pipelines on multiple Nextflow versions
                    null, // removeNextflowVersion("$outputDir/pipeline_info/phac_nml_metadatatransformation_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }
}
