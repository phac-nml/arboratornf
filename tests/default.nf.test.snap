{
    "-profile test": {
        "content": [
            9,
            null,
            [
                "arborator",
                "arborator/1_clusters.tsv",
                "arborator/1_loci.summary.tsv",
                "arborator/1_matrix.tsv",
                "arborator/1_metadata.tsv",
                "arborator/1_outliers.tsv",
                "arborator/1_profile.tsv",
                "arborator/1_tree.nwk",
                "arborator/2_clusters.tsv",
                "arborator/2_loci.summary.tsv",
                "arborator/2_matrix.tsv",
                "arborator/2_metadata.tsv",
                "arborator/2_outliers.tsv",
                "arborator/2_profile.tsv",
                "arborator/2_tree.nwk",
                "arborator/cluster_summary.tsv",
                "arborator/metadata.excluded.tsv",
                "arborator/metadata.included.tsv",
                "arborator/threshold_map.json",
                "arborview",
                "arborview/1_arborview.html",
                "arborview/2_arborview.html",
                "build",
                "build/config.json",
                "locidex",
                "locidex/concat",
                "locidex/concat/MLST_error_report_concat.csv",
                "locidex/concat/profile_concat.tsv",
                "locidex/merged",
                "locidex/merged/MLST_error_report_1.csv",
                "locidex/merged/profile_1.tsv",
                "metadata",
                "metadata/aggregated_data.tsv",
                "pipeline_info",
                "pipeline_info/software_versions.yml",
                "write",
                "write/results.csv"
            ],
            [
                "1_clusters.tsv:md5,d0e181df4d84b0bc7f774d098004e429",
                "1_loci.summary.tsv:md5,d4241e1324dddaab2a56adf2f2d04f22",
                "1_matrix.tsv:md5,450180a83f957d6660bdc4d4ca976097",
                "1_metadata.tsv:md5,da5ce3baf646490be90c6181506c2f46",
                "1_outliers.tsv:md5,c768621c9a61352f70fa9c26891e2c5a",
                "1_profile.tsv:md5,9ad1241e7131f6ab570d2f9efb85ed55",
                "1_tree.nwk:md5,d3b3527574f1d1dec01b5b4978f65002",
                "2_clusters.tsv:md5,055490c0e91d8eb50a2a367af92983dd",
                "2_loci.summary.tsv:md5,399ba4bd5afa213413b7eb261910f9f2",
                "2_matrix.tsv:md5,076f61db93ffdef096525a00a10133f0",
                "2_metadata.tsv:md5,24d68425c0d5a8576387ab9ff7a8c7ca",
                "2_outliers.tsv:md5,c768621c9a61352f70fa9c26891e2c5a",
                "2_profile.tsv:md5,41e92b4c54357b063fa5b10d35bb93b5",
                "2_tree.nwk:md5,fd4b70c4ad5c77f8047435678928fa97",
                "cluster_summary.tsv:md5,2178627f8fcea63b3c858868fd62e1d4",
                "metadata.excluded.tsv:md5,9e6f99c7155f3c4a30b3b80235b6af52",
                "metadata.included.tsv:md5,da1a072c2ed37bdc45c962ced9c01055",
                "threshold_map.json:md5,cc781d4410fdc58529d1aeb9513e132d",
                "1_arborview.html:md5,93902625588fedf18cce7b52a3e15912",
                "2_arborview.html:md5,8f9937555ee1cde31c5c03d961b8cef7",
                "config.json:md5,d7810954079f9f96c60466340a7c8f6c",
                "MLST_error_report_concat.csv:md5,f6689b46a8f71e85ee04acc80800ad84",
                "profile_concat.tsv:md5,1612b39148782f37b1e97d14ce72ac71",
                "MLST_error_report_1.csv:md5,f6689b46a8f71e85ee04acc80800ad84",
                "profile_1.tsv:md5,1612b39148782f37b1e97d14ce72ac71",
                "aggregated_data.tsv:md5,1c8602e7f3c69e54ac07f9d4008a20fe",
                "results.csv:md5,37930acecfc7ab97a314be0e859f6960"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-09-04T12:45:27.112752222"
    }
}